cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/MetadataSearchClient.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.capability;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.http.DefaultHttpRequestConfig;
+import io.cdap.cdap.common.internal.remote.RemoteClient;
+import io.cdap.cdap.proto.metadata.MetadataSearchResponse;
+import io.cdap.cdap.spi.metadata.SearchRequest;
+import io.cdap.common.http.HttpMethod;
+import io.cdap.common.http.HttpRequest;
+import io.cdap.common.http.HttpResponse;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+/**
+ * SearchClient for internal searching from metadata service
+ */
+public class MetadataSearchClient {
+
+  private static final String QUERY_PARAM_FORMAT = "%s=%s";
+  private static final String METADATA_SEARCH_BASE = "metadata/search";
+  public static final String PARAM_DELIMITER = "&";
+  public static final String PARAM_SEPARATOR = "?";
+  private final RemoteClient remoteClient;
+  private static final Gson GSON = new GsonBuilder().create();
```

Comment:
Group static fields together. Also, you can just `new Gson()` in this case.

Reply:
Done